graylog-storage-opensearch3/src/main/java/org/graylog/storage/opensearch3/MoreSearchAdapterOS.java

```diff
@@ -0,0 +1,362 @@
+import jakarta.inject.Named;
+import org.graylog.events.event.EventDto;
+import org.graylog.events.processor.EventProcessorException;
+import org.graylog.events.search.MoreSearch;
+import org.graylog.events.search.MoreSearchAdapter;
+import org.graylog.plugins.views.search.searchfilters.model.UsedSearchFilter;
+import org.graylog.plugins.views.search.searchtypes.pivot.buckets.AutoInterval;
+import org.graylog2.indexer.results.ChunkedResult;
+import org.graylog2.indexer.results.MultiChunkResultRetriever;
+import org.graylog2.indexer.results.ResultChunk;
+import org.graylog2.indexer.results.ResultMessage;
+import org.graylog2.indexer.searches.ChunkCommand;
+import org.graylog2.indexer.searches.Sorting;
+import org.graylog2.plugin.Message;
+import org.graylog2.plugin.Tools;
+import org.graylog2.plugin.indexer.searches.timeranges.AbsoluteRange;
+import org.graylog2.plugin.indexer.searches.timeranges.TimeRange;
+import org.opensearch.client.json.JsonData;
+import org.opensearch.client.opensearch._types.ExpandWildcard;
+import org.opensearch.client.opensearch._types.FieldSort;
+import org.opensearch.client.opensearch._types.FieldValue;
+import org.opensearch.client.opensearch._types.SortOptions;
+import org.opensearch.client.opensearch._types.SortOrder;
+import org.opensearch.client.opensearch._types.aggregations.Aggregation;
+import org.opensearch.client.opensearch._types.aggregations.DateHistogramAggregate;
+import org.opensearch.client.opensearch._types.aggregations.DateHistogramBucket;
+import org.opensearch.client.opensearch._types.aggregations.FieldDateMath;
+import org.opensearch.client.opensearch._types.aggregations.LongTermsAggregate;
+import org.opensearch.client.opensearch._types.aggregations.MultiBucketBase;
+import org.opensearch.client.opensearch._types.mapping.FieldType;
+import org.opensearch.client.opensearch._types.query_dsl.BoolQuery;
+import org.opensearch.client.opensearch._types.query_dsl.Query;
+import org.opensearch.client.opensearch._types.query_dsl.TermsQueryField;
+import org.opensearch.client.opensearch.core.SearchResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+public class MoreSearchAdapterOS implements MoreSearchAdapter {
+    private static final Logger LOG = LoggerFactory.getLogger(MoreSearchAdapterOS.class);
+    private static final String termsAggregationName = "alert_type";
+    private static final String histogramAggregationName = "histogram";
```

Comment:
Could those constants be moved to the interface?
Shouldn't they be written with capital letters to indicate they are constants?